### PR TITLE
Commit flags

### PR DESCRIPTION
**Pull request by mfocko**
- [x] add methods
- [x] services' implementation
- [x] enum for statuses?
- [x] PRs?
  - [x] tests

Comparison of statuses:

GitHub | GitLab | Pagure
--- | --- | ---
`pending` | `pending` | `pending`
`success` | `success` | `success`
`failure` | `failed` | `failure`
`error` | `-` | `error`
`-` | `canceled` | `canceled`
`-` | `running` | `-`

### CODE REVIEW

File: ogr/abstract.py [hunk 1]
```diff
@@ -437,35 +437,84 @@ def add_label(self, *labels: str) -> None:
...
+    def get_statuses(self) -> List["CommitFlag"]:
+        """
+        Returns statuses for latest commit on pull request.
+
+        :return: [CommitFlag]
+        """
+        raise NotImplementedError()
+
+
+class CommitStatus(IntEnum):
```
Comment (lachmanfrantisek) [inline ogr/abstract.py]: Do we need the comparison with `int`s?

What about this?
https://docs.python.org/3/library/enum.html#using-a-descriptive-string

Comment (mfocko) [reply ogr/abstract.py]: Had a look around abstract, I thought it's a convention, ok, I'll change it

Comment (lachmanfrantisek) [reply ogr/abstract.py]: It allows some `string <-> enum` conversion.

**Pull request merged**